pinot-core/src/main/java/org/apache/pinot/core/operator/query/SelectionOnlyOperator.java

```diff
@@ -75,6 +75,8 @@ public SelectionOnlyOperator(IndexSegment indexSegment, QueryContext queryContex
     _dataSchema = new DataSchema(columnNames, columnDataTypes);
 
     _numRowsToKeep = queryContext.getLimit();
+    // TODO(gortiz): I think this is incorrect. The SelectionOperatorUtils.MAX_ROW_HOLDER_INITIAL_CAPACITY limit
+    //  is not enforced later in getNextBlock
```

Comment:
this hint is only used during leaf initialization. i don't think there's any plan to create another default on intermediate select/project. but we can see if that helps

Reply:
We need to limit the initial list size because it might not be filled up. In a lot of scenario `LIMIT` can be very large (could be Integer.MAX_VALUE)